service/src/main/java/bio/terra/workspace/app/controller/WorkspaceV2Api.java

```diff
@@ -0,0 +1,222 @@
+  }
+
+  public ResponseEntity<ApiCreateWorkspaceV2Result> createWorkspaceV2(
+      ApiCreateWorkspaceV2Request body) {
+    AuthenticatedUserRequest userRequest = authenticatedUserRequestFactory.from(request);
+    logger.info(
+        "Creating workspace {} and {} cloud context for {} subject {}",
+        body.getId(),
+        (body.getCloudPlatform() == null ? "no" : body.getCloudPlatform()),
+        userRequest.getEmail(),
+        userRequest.getSubjectId());
+    String jobId = body.getJobControl().getId();
+
+    // If we are creating a cloud context (cloudPlatform is present),
+    // then the spend profile must be valid and the user must have
+    // permission to link to it.
+    CloudPlatform cloudPlatform = null;
+    SpendProfile spendProfile = null;
+    SpendProfileId spendProfileId = null;
+    ApiCloudPlatform apiCloudPlatform = body.getCloudPlatform();
+    if (apiCloudPlatform != null) {
+      ControllerValidationUtils.validateCloudPlatform(body.getCloudPlatform());
+      cloudPlatform = CloudPlatform.fromApiCloudPlatform(apiCloudPlatform);
+      spendProfile =
+          workspaceApiUtils.validateSpendProfilePermission(userRequest, body.getSpendProfile());
+      if (spendProfile == null) {
+        throw new MissingRequiredFieldsException(
+            "To create a cloud context you must provide both cloudPlatform and spendProfile");
+      }
+      spendProfileId = spendProfile.id();
+    }
+
+    TpsPolicyInputs policies =
+        workspaceApiUtils.validateAndConvertPolicies(body.getPolicies(), body.getStage());
+    // WSM requires a userFacingId. Create one, if it is not provided.
+    WorkspaceStage workspaceStage = WorkspaceApiUtils.getStageFromApiStage(body.getStage());
+    String userFacingId =
+        Optional.ofNullable(body.getUserFacingId()).orElse(body.getId().toString());
+    ControllerValidationUtils.validateUserFacingId(userFacingId);
+
+    Workspace workspace =
+        Workspace.builder()
+            .workspaceId(body.getId())
+            .userFacingId(userFacingId)
+            .displayName(body.getDisplayName())
+            .description(body.getDescription())
+            .spendProfileId(spendProfileId)
+            .workspaceStage(workspaceStage)
+            .properties(convertApiPropertyToMap(body.getProperties()))
+            .createdByEmail(samService.getUserEmailFromSamAndRethrowOnInterrupt(userRequest))
+            .build();
+
+    workspaceService.createWorkspaceV2(
+        workspace,
+        policies,
+        /*applications=*/ null,
+        cloudPlatform,
+        spendProfile,
+        jobId,
+        userRequest);
+
+    ApiCreateWorkspaceV2Result result = fetchCreateWorkspaceV2Result(jobId);
+    return new ResponseEntity<>(
+        result, ControllerUtils.getAsyncResponseCode(result.getJobReport()));
+  }
+
+  private ApiCreateWorkspaceV2Result fetchCreateWorkspaceV2Result(String jobId) {
+    JobApiUtils.AsyncJobResult<Workspace> jobResult =
```

Comment:
Since we have no surrounding container controlling access, what permission would I check?
We will probably have one in the future, but not now. Do you see any other choice but reverting to the V1 behavior and returning just the workspace id?

Reply:
I'm doing #3

Reply:
I think I will go ahead and do 3.